version: '3.8'
services:
  web:

    build: .
    ports:
      - "8080:8080"
    depends_on:
      - db
    environment:
          SPRING_DATASOURCE_URL: jdbc:mysql://db:3307/newtest_db?autoReconnect=true&useSSL=false&allowPublicKeyRetrieval=true
    networks:
      - spring-cloud-network


  db:
    image: mysql:8.0
    cap_add:
      - SYS_NICE
    restart: always
    environment:
      - MYSQL_DATABASE=newtest_db
      - MYSQL_ROOT_USERNAME=root
      - MYSQL_ROOT_PASSWORD=password
      - MYSQL_TCP_PORT=3307
    ports:
      - '3307:3307'
    expose:
      - 3307
    volumes:
      - db:/var/lib/mysql
      - ./db/init.sql:/docker-entrypoint-initdb.d/init.sql
    networks:
      - spring-cloud-network
    healthcheck:
      test: [ "CMD", "curl", "-f", "http://localhost:3307" ]
      interval: 30s
      timeout: 10s
      retries: 5

volumes:
  db:
    driver: local

networks:
  spring-cloud-network:
    driver: bridge